contracts/colony/Colony.sol

```diff
@@ -37,6 +37,25 @@ contract Colony is ColonyStorage, PatriciaTreeProofs {
     return token;
   }
 
+  bytes4 constant TRANSFER_SIG = bytes4(keccak256("transfer(address,uint256)"));
+  bytes4 constant TRANSFER_FROM_SIG = bytes4(keccak256("transferFrom(address,address,uint256)"));
+
+  function makeArbitraryTransaction(address _to, uint256 _value, bytes memory _action)
+  public stoppable auth
+  returns (bool)
+  {
+    require(_to != colonyNetworkAddress, "colony-cannot-target-network");
+    require(_to != tokenLockingAddress, "colony-cannot-target-token-locking");
+
+    bytes4 sig;
+    assembly { sig := mload(add(_action, 0x20)) }
+
+    require(sig != TRANSFER_SIG, "colony-cannot-call-erc20-transfer");
+    require(sig != TRANSFER_FROM_SIG, "colony-cannot-call-erc20-transfer-from");
```

Comment:
I don't follow why `transferFrom` needs to be restricted? Did you mean `approve`?

Reply:
Couldn't a colony call `transferFrom` and pass itself in as an argument, to get around the restriction on `transfer`?

Reply:
Only if it's given itself approval, assuming it's a well behaved ERC20. At the very least, we need to prevent it giving other addresses approval to take from it.

Reply:
Ah fair point. I'll add the restriction on `approve`. Part of me likes leaving in the restriction on `transferFrom` for completeness, but I suppose it's not strictly necessary.

Reply:
I don't think it's right. A colony might well want to call `transferFrom` if it is given control over another's tokens.